plugins/woocommerce/client/blocks/assets/js/blocks/product-reviews/inner-blocks/reviews-title/edit.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { useEntityProp } from '@wordpress/core-data';
+import { PanelBody, ToggleControl } from '@wordpress/components';
+import { useState, useEffect } from '@wordpress/element';
+import { resolveSelect } from '@wordpress/data';
+import { reviewsStore } from '@woocommerce/data';
+import {
+	// @ts-expect-error AlignmentControl is not exported from @wordpress/block-editor
+	AlignmentControl,
+	BlockControls,
+	useBlockProps,
+	InspectorControls,
+	// @ts-expect-error HeadingLevelDropdown is not exported from @wordpress/block-editor
+	HeadingLevelDropdown,
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import type { ProductReviewsTitleEditProps } from './types';
+
+export default function Edit( {
+	attributes: {
+		textAlign,
+		showProductTitle,
+		showReviewsCount,
+		level,
+		levelOptions,
+	},
+	setAttributes,
+	context: { postType, postId },
+}: ProductReviewsTitleEditProps ) {
+	const TagName = 'h' + level;
+	const [ reviewsCount, setReviewsCount ] = useState< number >( 3 );
+	const [ rawTitle ] = useEntityProp( 'postType', postType, 'title', postId );
+	const isSiteEditor = typeof postId === 'undefined';
+	const blockProps = useBlockProps( {
+		className: clsx( {
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+		} ),
+	} );
+
+	useEffect( () => {
+		if ( isSiteEditor ) {
+			setReviewsCount( 3 );
+			return;
+		}
+		resolveSelect( reviewsStore )
+			.getReviewsTotalCount( {
```

Comment:
Had to use the `reviewsStore` for the data as the WP comments API doesn't work for this.
This is the main reason I created a new block for this.